src/language-js/printer-estree.js

```diff
@@ -5128,6 +5129,60 @@ function printMemberChain(path, options, print) {
   ]);
 }
 
+/**
+ * @param {import('estree').Node} node
+ * @returns {boolean}
+ */
+function isSimple(node, depth = 0) {
+  if (depth >= 2) {
+    return false;
+  }
+  const isChildSimple = child => isSimple(child, depth + 1);
+  if (
+    !node ||
+    node.type === "Literal" ||
+    node.type === "BooleanLiteral" ||
+    node.type === "NullLiteral" ||
+    node.type === "NumericLiteral" ||
+    node.type === "StringLiteral" ||
+    node.type === "Identifier" ||
+    node.type === "ThisExpression" ||
+    node.type === "Super" ||
+    node.type === "BigIntLiteral" ||
+    node.type === "PrivateName" ||
+    node.type === "ArgumentPlaceholder" ||
+    node.type === "RegExpLiteral"
+  ) {
+    return true;
+  }
+  if (node.type === "TemplateLiteral") {
+    return node.expressions.every(isChildSimple);
+  }
+  if (node.type === "ObjectExpression") {
+    return node.properties.every(p => p.shorthand);
+  }
+  if (node.type === "ArrayExpression") {
+    return node.elements.every(isChildSimple);
+  }
+  if (
+    node.type === "CallExpression" ||
+    node.type === "OptionalCallExpression" ||
+    node.type === "NewExpression"
+  ) {
+    return isSimple(node.callee) && node.arguments.every(isChildSimple);
```

Comment:
`isSimple` resets `depth` to `0`. Think of deeply nested calls like `foo(bar(baz(qux(), quux()))`. Do they look simple?